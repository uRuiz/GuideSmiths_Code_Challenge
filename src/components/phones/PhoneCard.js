import React from 'react'
import { Link } from 'react-router-dom'

export const PhoneCard = ({ imageFileName, name, price, id}) => {
    return (
        <div className="card ms-3" style={{ maxWidth: 500 }}>
            <div className="row no-gutters">
                <img src={`./assets/images/${imageFileName}`} className="card-img" alt={name} />
            </div>
            <div className="col-md-8">
                <div className="card-body">
                    <h5 className="card-title">{name}</h5>
                    <p>{price}</p>
                    <Link to={`./phone/${id}`}>
                        Full Info...
                    </Link>
                </div>
            </div>
        </div>
    )
}
