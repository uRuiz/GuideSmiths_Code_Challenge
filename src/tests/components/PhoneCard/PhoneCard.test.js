import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { PhoneCard } from '../../../components/Phones/PhoneCard';

describe('Testing <PhoneCard /> component', () => {

    test('should show <PhoneCard /> as expected', () => {
        
        const wrapper = shallow(<PhoneCard 
                id={0}
                imageFileName='IPhone_7.png'
                key={0}
                name='iPhone 7'
                price={769}
        />);

        expect( wrapper ).toMatchSnapshot();

    });   
    
})