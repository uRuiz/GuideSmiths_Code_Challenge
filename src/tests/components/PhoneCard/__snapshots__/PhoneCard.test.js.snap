// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <PhoneCard /> component should show <PhoneCard /> as expected 1`] = `
<div
  className="card ms-3"
  style={
    Object {
      "maxWidth": 500,
    }
  }
>
  <div
    className="row no-gutters"
  >
    <Link
      className="col-md-12"
      to="./phone/id=0"
    >
      <img
        alt="iPhone 7"
        className="card-img"
        src="./assets/images/IPhone_7.png"
      />
    </Link>
  </div>
  <div
    className="col-md-12"
  >
    <div
      className="card-body"
    >
      <h5
        className="card-title"
      >
        iPhone 7
      </h5>
      <p>
        769
         â‚¬
      </p>
      <Link
        to="./phone/id=0"
      >
        Full Info...
      </Link>
    </div>
  </div>
</div>
`;
